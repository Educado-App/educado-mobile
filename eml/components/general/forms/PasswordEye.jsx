import React from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import PropTypes from 'prop-types';
import { Pressable } from 'react-native';

/**
 * Icon component for the eye besides passwords to toggle if text can be seen by user
 * @param {Object} props Should contain the following properties
 * - showPasswordIcon: Boolean
 * - toggleShowPassword: Function
 */
export default function PasswordEye(props) {
	return (
		<Pressable
			className="absolute right-0 top-2 p-3"
			onPress={props.toggleShowPassword}
		>
			<MaterialCommunityIcons
				name={props.showPasswordIcon ? 'eye-off' : 'eye'}
				size={24}
				color="gray"
			/>
		</Pressable>
	);
}

PasswordEye.propTypes = {
	showPasswordIcon: PropTypes.bool,
	toggleShowPassword: PropTypes.func,
};