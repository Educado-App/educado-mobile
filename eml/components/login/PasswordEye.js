import React from 'react';
import { View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

export default function PasswordEye(props) {

  return (
    <View className='absolute right-2 top-2 mr-10 mt-3'>
      <MaterialCommunityIcons
        name={props.showPasswordIcon ? 'eye-off' : 'eye'}
        size={24}
        color="#aaa"
        onPress={props.toggleShowPassword}
      />
    </View>
  )
}