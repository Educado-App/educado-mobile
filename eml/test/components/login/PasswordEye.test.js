import renderer from "react-test-renderer";
import React from "react";
import PasswordEye from "../../../components/general/forms/PasswordEye";

const mockFunction = jest.fn();

test('Ensure that the field is rendered correctly without props', async () => {
  let passwordEye;
  await renderer.act(() => {
    passwordEye = renderer.create(<PasswordEye />);
  });
  let tree = passwordEye.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Ensure that the field props are rendered correctly with props', async () => {
  let passwordEye;
  await renderer.act(() => {
    passwordEye = renderer.create(<PasswordEye 
      name="eye" 
      size={24} 
      color="gray"
      onPress={mockFunction}
    />);
  });
  expect(passwordEye.root.props.name).toBe('eye');
  expect(passwordEye.root.props.size).toBe(24);
  expect(passwordEye.root.props.color).toBe('gray');
  expect(passwordEye.root.props.onPress).toBe(mockFunction);
});