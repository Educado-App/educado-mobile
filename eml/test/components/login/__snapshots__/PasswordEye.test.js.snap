// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the field is rendered correctly without props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "position": "absolute",
      },
      {
        "right": 0,
      },
      {
        "top": 8,
      },
      {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "gray",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ó°ˆˆ
  </Text>
</View>
`;
